### PR TITLE
Draft change to report in case no buffer available

### PR DESCRIPTION
**Pull request by lyjbupt**

### CODE REVIEW

File: omx_components/src/mfx_omx_vdec_component.cpp [hunk 1]
```diff
@@ -2033,7 +2033,12 @@ void MfxOmxVdecComponent::MainThread(void)
                             }
                         }
                         else
+                        { 
+                            MFX_OMX_LOG_INFO_IF(g_OmxLogLevel, "Failed to get buffer, maybe all buffers are allocated!");
```
Comment (DenWolf) [inline omx_components/src/mfx_omx_vdec_component.cpp]: There is no error, branch "else" is correct in this way.
We can get here when we decoded full bitstream (which we've got from framework), and we are going to wait new part of bitstream from framework to continues decoding

**Review by DenWolf — CHANGES_REQUESTED** [review]
👎

File: omx_components/src/mfx_omx_vdec_component.cpp (hunk 1 above)
Comment (lyjbupt) [reply omx_components/src/mfx_omx_vdec_component.cpp]: Revert changes back to initial PR and also add brief error handing, please check if this is fine?

File: omx_buffers/src/mfx_omx_srf_pool.cpp [hunk 2]
```diff
@@ -254,6 +254,7 @@ mfxStatus MfxOmxSurfacesPool::FindAvailableSurface(OMX_BUFFERHEADERTYPE** ppBuff
 
     if (!bFound)
     {
+	MFX_OMX_LOG_ERROR(g_OmxLogLevel, "Failed to find output surface, maybe all buffers are allocated!");
```
Comment (DenWolf) [inline omx_buffers/src/mfx_omx_srf_pool.cpp]: First, buffers allocated in another component, Media SDK OMX handles already allocated buffers. so comment not fully correct.
Function FindAvailableSurface(...) is called from UseBuffer(...) function (below), and tries to find available (non-locked) surface.

The main question from me - what exactly do you want to achieve by this change ?

Comment (lyjbupt) [reply omx_buffers/src/mfx_omx_srf_pool.cpp]: Upload new changes

File: omx_components/src/mfx_omx_vdec_component.cpp [hunk 3]
```diff
@@ -617,7 +617,11 @@ OMX_ERRORTYPE MfxOmxVdecComponent::DealWithBuffer(
                 m_pSurfaces->SetMaxErrorCount(port_info->nCurrentBuffersCount+1);
             }
         }
-        else mfx_sts = MFX_ERR_MEMORY_ALLOC;
+        else 
+	{
+	    MFX_OMX_LOG_ERROR(g_OmxLogLevel, "Failed to allocate buffers, maybe all buffers are allocated!");
```
Comment (TianmiChen) [inline omx_components/src/mfx_omx_vdec_component.cpp]: Is it better to add the information of these buffers (pBufferHeader, pBuffer)? For example,
MFX_OMX_LOG_ERROR(g_OmxLogLevel, "Failed to allocate the buffer, pBufferHeader=%p, pBuffer=%p", pBufferHeader, pBuffer);

**Pull request merged**